src/System.Management.Automation/FormatAndOutput/common/BaseCommand.cs

```diff
@@ -251,20 +251,24 @@ protected virtual void Dispose(bool disposing)
         {
             if (disposing)
             {
+                if (this.implementation != null)
+                {
+                    this.implementation.Dispose();
+                    this.implementation = null;
+                }
+
+#pragma warning disable CS0618 // Type or member is obsolete - Call is required for backwards compatibility
                 InternalDispose();
+#pragma warning restore CS0618 // Type or member is obsolete - Call is required for backwards compatibility
             }
         }
 
         /// <summary>
         /// Do-nothing implementation: derived classes will override as see fit.
         /// </summary>
+        [Obsolete("Derived classes should override Dispose(bool disposing) instead.", false)]
         protected virtual void InternalDispose()
```

Comment:
Since the `FrontEndCommandBase` class is public, I left this method in for compatibility with potential users. I suppose it could be a good idea to deprecate it now and remove it later?